### DISCUSSION TITLE
Issue with GPU parallelization in LAMMPS from conda-built deepmd-gpu-2.2.10

### DISCUSSION THREAD

**Discussion by chenggoj** [deepmodeling/deepmd-kit#4170] [Q&A]
Hi，
I've set up an environment using conda with deepmd-gpu-2.2.10 and dpgen. When training with DeePMD-kit using `CUDA_VISIBLE_DEVICES=0,1,2,3 mpirun -l -launcher=fork -hosts=localhost -np 4 dp`, all 4 GPUs are fully utilized.
However, when running LAMMPS with the default `lmp` command, only one GPU is used. Attempting to use `CUDA_VISIBLE_DEVICES=0,1,2,3 mpirun -np 4 lmp` fails to execute the LAMMPS task.

Questions:

Does the default LAMMPS in the conda-built deepmd-gpu-2.2.10 not support MPI parallelization?
But I noticed that `lmp` should be compiled with MPI support when I run

` lmp -h `

> Compiler: GNU C++ 7.5.0 with OpenMP 4.5
> C++ standard: C++11
> MPI v4.0: MPICH Version:      4.1.1
> MPICH Release date: Mon Mar  6 14:14:15 CST 2023
> MPICH ABI:          15:0:3

Is GPU acceleration not supported in this LAMMPS build? (I noticed the GPU package is not listed in the Installed packages)

> Installed packages:
> 
> ASPHERE BODY CLASS2 COLLOID COLVARS CORESHELL DIPOLE EXTRA-COMPUTE EXTRA-DUMP 
> EXTRA-FIX EXTRA-MOLECULE EXTRA-PAIR FEP GRANULAR H5MD KSPACE MANYBODY MC MEAM 
> MISC MOLECULE OPT PERI PHONON PLUGIN PLUMED QTB REAXFF REPLICA RIGID SHOCK SRD
> 

How can I enable multi-GPU support for LAMMPS in this environment? Do I have to rebuild and install LAMMPS from the source code with build-in or plugin from DeePMD-kit? My ultimate goal is to fully utilize all GPU resources by running `lmp` in parallel on GPU nodes.

Also, I've spent considerable time trying to compile LAMMPS from source, but have consistently failed due to various library or version incompatibility issues. This has been extremely frustrating. Any guidance on properly configuring LAMMPS for multi-GPU usage within the conda-built 

> deepmd-gpu-2.2.10 

environment would be greatly appreciated.

**Comment by rashidrafeek** [comment]
Was this issue resolved. Were you able to run lammps with multiple GPUs?
